ethsigner/commandline/src/main/java/tech/pegasys/ethsigner/config/tls/client/PicoCliClientTlsOptions.java

```diff
@@ -64,4 +69,49 @@ public boolean isCaAuthEnabled() {
   public Optional<Path> getKnownServersFile() {
     return Optional.ofNullable(knownServersFile);
   }
+
+  public boolean isTlsEnabled() {
+    return tlsEnabled;
+  }
+
+  public String validationMessage() {
+    if (!tlsEnabled) {
+      return dependentOptionSpecifiedMessage();
+    }
+
+    return keyStoreConfigurationValidationMessage() + knownServerFileAndCaValidationMessage();
+  }
+
+  private String dependentOptionSpecifiedMessage() {
+    if (knownServersFile != null
+        || keyStoreOptions.getKeyStoreFile() != null
+        || keyStoreOptions.getPasswordFile() != null) {
+      return "Missing required argument(s): '--downstream-http-tls-enabled=true'\n";
+    }
+    return "";
+  }
+
+  private String knownServerFileAndCaValidationMessage() {
+    if (knownServersFile == null && !caAuthEnabled) {
+      return "Missing required argument(s): '--downstream-http-tls-known-servers-file' must be specified if '--downstream-http-tls-ca-auth-enabled=false'\n";
+    }
+    return "";
+  }
+
+  private String keyStoreConfigurationValidationMessage() {
+    final List<String> missingOptions = new ArrayList<>();
+
+    // ArgGroup custom validation
+    if (keyStoreOptions.getKeyStoreFile() == null && keyStoreOptions.getPasswordFile() != null) {
+      missingOptions.add("'--downstream-http-tls-keystore-file=" + FILE_FORMAT_HELP + "'");
+    } else if (keyStoreOptions.getKeyStoreFile() != null
+        && keyStoreOptions.getPasswordFile() == null) {
+      missingOptions.add("'--downstream-http-tls-keystore-password-file=" + FILE_FORMAT_HELP + "'");
+    }
+
+    if (!missingOptions.isEmpty()) {
+      return "Missing required arguments(s): " + String.join(",", missingOptions) + "\n";
```

Comment:
to be platform agnostic - you need to do a String.format with "%n" rather than just using \n

Reply:
good catch, moved to using System.lineSeperator.